identity/engines/pkcs11/engine.go

```diff
@@ -162,8 +163,17 @@ func (*engine) LoadKey(key *url.URL) (crypto.PrivateKey, error) {
 			log.Warnf("slot not specified, using first slot reported by the driver (%d)", slotId)
 		}
 	} else {
-		id, _ := strconv.Atoi(slot)
-		slotId = uint(id)
+		if strings.HasPrefix(slot, "0x") {
+			n, err := strconv.ParseInt(slot, 0, 64)
```

Comment:
Doesn't ParseInt inspect the prefix for you when you pass `0` for the base?
I would think you could just use ParseInt (or ParseUInt) without looking for the "0x" prefix.